point_viewer_grpc/src/bin/octree_benchmark.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#[macro_use]
+extern crate clap;
+extern crate futures;
+extern crate grpcio;
+extern crate point_viewer;
+extern crate point_viewer_grpc;
+extern crate point_viewer_grpc_proto_rust;
+
+use point_viewer_grpc_proto_rust::proto;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use futures::future::Future;
+use futures::Stream;
+use grpcio::{ChannelBuilder, Environment};
+use point_viewer::octree::OnDiskOctree;
+use point_viewer::{InternalIterator, Point};
+use point_viewer_grpc::proto_grpc::OctreeClient;
+use point_viewer_grpc::service::start_grpc_server;
+
+fn main() {
+    let matches = clap::App::new("octree_benchmark")
+        .args(&[
+            clap::Arg::with_name("port")
+                .help("Port for the server to listen on for connections. [50051]")
+                .long("port")
+                .takes_value(true),
+            clap::Arg::with_name("no-client")
+                .help("Do not actually send points, only read them on the server.")
+                .long("no-client")
+                .takes_value(false),
+            clap::Arg::with_name("num-points")
+                .help("Number of points to stream. [50000000]")
+                .long("num-points")
+                .takes_value(true),
+            clap::Arg::with_name("octree_directory")
+                .help("Input directory of the octree directory to serve.")
+                .index(1)
+                .required(true),
+        ])
+        .get_matches();
+
+    let octree_directory = PathBuf::from(
+        matches
+            .value_of("octree_directory")
+            .expect("octree_directory not given"),
+    );
+    let num_points = u64::from_str(matches.value_of("num-points").unwrap_or("50000000"))
+        .expect("Could not num-points");
+    if matches.is_present("no-client") {
+        server_benchmark(octree_directory, num_points)
+    } else {
+        let port = value_t!(matches, "port", u16).unwrap_or(50051);
+        full_benchmark(octree_directory, num_points, port)
+    }
+}
+
+fn server_benchmark(octree_directory: PathBuf, num_points: u64) {
+    let octree = OnDiskOctree::new(&octree_directory).unwrap();
```

Comment:
do an `.expect("proper error message")` instead of `.unwrap()`

Reply:
Done